segfault in builder/amazon-ebs

don't have a json that will reproduce this yet, but shouldn't be too hard. Thinking it might have been introduced in 46f217f255e81059baace087a12fb8dd202078b9

```==> aws: Waiting for AMI to become ready... 
==> aws: Terminating the source AWS instance... 
==> aws: Cleaning up any extra volumes... 
==> aws: No volumes to clean up, skipping 
==> aws: Deleting temporary security group... 
==> aws: Deleting temporary keypair... 
Build 'aws' errored: unexpected EOF

==> Some builds didn't complete successfully and had errors: 
--> aws: unexpected EOF

==> Builds finished but no artifacts were created. 
panic: runtime error: invalid memory address or nil pointer dereference 
2016/12/20 02:50:08 packer: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x5ecc57] 
2016/12/20 02:50:08 packer: 
2016/12/20 02:50:08 packer: goroutine 76 [running]: 
2016/12/20 02:50:08 packer: panic(0x13beea0, 0xc42000e0b0) 
2016/12/20 02:50:08 packer: /usr/local/Cellar/go/1.7.3/libexec/src/runtime/panic.go:500 +0x1a1 
2016/12/20 02:50:08 packer: 
github.com/mitchellh/packer/builder/amazon/ebs.(*stepCreateAMI).Run(0xc4200240c0 
, 0x1e117a0, 0xc420363cb0, 0x0) 
2016/12/20 02:50:08 packer: /Users/mwhooker/go/src/github.com/mitchellh/packer/builder/amazon/ebs/step_create_ami.go:72 +0x917 
2016/12/20 02:50:08 packer: 
github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep.(*BasicRunner).Run(0xc420434100 
, 0x1e117a0, 0xc420363cb0) 
2016/12/20 02:50:08 packer: /Users/mwhooker/go/src/github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep/basic_runner.go:70 +0x207 
2016/12/20 02:50:08 packer: 
github.com/mitchellh/packer/builder/amazon/ebs.(*Builder).Run(0xc42020a000 
, 0x1e140e0, 0xc4202b9540, 0x1e0d0a0, 0xc4202c07b0, 0x1e12e60, 0xc420024490, 0xc42029fe30, 0x0, 0x0, ...) 
2016/12/20 02:50:08 packer: /Users/mwhooker/go/src/github.com/mitchellh/packer/builder/amazon/ebs/builder.go:196 +0x11cb 
2016/12/20 02:50:08 packer: 
github.com/mitchellh/packer/packer/rpc.(*BuilderServer).Run(0xc420228000 
, 0x1, 0xc4202c0710, 0x0, 0x0) 
2016/12/20 02:50:08 packer: /Users/mwhooker/go/src/github.com/mitchellh/packer/packer/rpc/builder.go:93 +0x218 
2016/12/20 02:50:08 packer: reflect.Value.call(0xc420013980, 0xc4200243f8, 0x13, 0x15ffd31, 0x4, 0xc4200b9ed0, 0x3, 0x3, 0x0, 0x0, ...) 
2016/12/20 02:50:08 packer: /usr/local/Cellar/go/1.7.3/libexec/src/reflect/value.go:434 +0x5c8 
2016/12/20 02:50:08 packer: reflect.Value.Call(0xc420013980, 0xc4200243f8, 0x13, 0xc4200b9ed0, 0x3, 0x3, 0x0, 0x0, 0x0) 
2016/12/20 02:50:08 packer: /usr/local/Cellar/go/1.7.3/libexec/src/reflect/value.go:302 +0xa4 
2016/12/20 02:50:08 packer: net/rpc.(*service).call(0xc420119f00, 0xc420119ec0, 0xc4201a9e00, 0xc420065f00, 0xc420228460, 0x13117c0, 0xc4202c070c, 0x18a, 0x12e6d00, 0xc4202c0710, ...) 
2016/12/20 02:50:08 packer: /usr/local/Cellar/go/1.7.3/libexec/src/net/rpc/server.go:383 +0x148 
2016/12/20 02:50:08 packer: created by net/rpc.(*Server).ServeCodec 
2016/12/20 02:50:08 packer: /usr/local/Cellar/go/1.7.3/libexec/src/net/rpc/server.go:477 +0x421 
2016/12/20 02:50:08 ui error: 
Build 'aws' errored: unexpected EOF

2016/12/20 02:50:08 Builds completed. Waiting on interrupt barrier... 
2016/12/20 02:50:08 machine readable: error-count []string{"1"} 
2016/12/20 02:50:08 ui error: 
==> Some builds didn't complete successfully and had errors: 
2016/12/20 02:50:08 machine readable: aws,error []string{"unexpected EOF"} 
2016/12/20 02:50:08 ui error: --> aws: unexpected EOF 
2016/12/20 02:50:08 ui: 
==> Builds finished but no artifacts were created. 
2016/12/20 02:50:08 waiting for all plugin processes to complete... 
2016/12/20 02:50:08 /usr/local/bin/packer: plugin process exited 
2016/12/20 02:50:08 /usr/local/bin/packer: plugin process exited 
2016/12/20 02:50:08 /usr/local/bin/packer: plugin process exited 
2016/12/20 02:50:08 /usr/local/bin/packer: plugin process exited 
2016/12/20 02:50:08 /usr/local/bin/packer: plugin process exited 
2016/12/20 02:50:08 /usr/local/bin/packer: plugin process exited

!!!!!!!!!!!!!!!!
```

Seeing a pretty similar crash with 0.12.1 as well:

```
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: Destroying volume (vol-082d871555c3e0922)...
==> amazon-ebs: Deleting temporary security group...
==> amazon-ebs: Deleting temporary keypair...
Build 'amazon-ebs' errored: unexpected EOF

==> Some builds didn't complete successfully and had errors:
--> amazon-ebs: unexpected EOF

==> Builds finished but no artifacts were created.
panic: runtime error: invalid memory address or nil pointer dereference
2016/12/21 00:21:24 packer: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1dab87]
2016/12/21 00:21:24 packer:
2016/12/21 00:21:24 packer: goroutine 9 [running]:
2016/12/21 00:21:24 packer: panic(0xfc55e0, 0xc420018120)
2016/12/21 00:21:24 packer: 	/usr/local/Cellar/go/1.7.4/libexec/src/runtime/panic.go:500 +0x1a1
2016/12/21 00:21:24 packer: github.com/mitchellh/packer/builder/amazon/ebs.(*stepCreateAMI).Run(0xc420030378, 0x1a21980, 0xc420244e40, 0x0)
2016/12/21 00:21:24 packer: 	/private/tmp/packer-20161216-4826-3b8w9v/src/github.com/mitchellh/packer/builder/amazon/ebs/step_create_ami.go:72 +0x917
2016/12/21 00:21:24 packer: github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep.(*BasicRunner).Run(0xc420163dc0, 0x1a21980, 0xc420244e40)
2016/12/21 00:21:24 packer: 	/private/tmp/packer-20161216-4826-3b8w9v/src/github.com/mitchellh/packer/vendor/github.com/mitchellh/multistep/basic_runner.go:70 +0x207
2016/12/21 00:21:24 packer: github.com/mitchellh/packer/builder/amazon/ebs.(*Builder).Run(0xc420272000, 0x1a24220, 0xc4201c9680, 0x1a1d280, 0xc4201d1a80, 0x1a23000, 0xc420030330, 0xc4202447b0, 0x0, 0x0, ...)
2016/12/21 00:21:24 packer: 	/private/tmp/packer-20161216-4826-3b8w9v/src/github.com/mitchellh/packer/builder/amazon/ebs/builder.go:196 +0x11cb
2016/12/21 00:21:24 packer: github.com/mitchellh/packer/packer/rpc.(*BuilderServer).Run(0xc42025ab60, 0x1, 0xc4201d19e0, 0x0, 0x0)
2016/12/21 00:21:24 packer: 	/private/tmp/packer-20161216-4826-3b8w9v/src/github.com/mitchellh/packer/packer/rpc/builder.go:93 +0x218
2016/12/21 00:21:24 packer: reflect.Value.call(0xc420260900, 0xc4202580b0, 0x13, 0x1206df7, 0x4, 0xc4206b3ed0, 0x3, 0x3, 0x0, 0x0, ...)
2016/12/21 00:21:24 packer: 	/usr/local/Cellar/go/1.7.4/libexec/src/reflect/value.go:434 +0x5c8
2016/12/21 00:21:24 packer: reflect.Value.Call(0xc420260900, 0xc4202580b0, 0x13, 0xc4206b3ed0, 0x3, 0x3, 0x16506, 0xc420380778, 0x1)
2016/12/21 00:21:24 packer: 	/usr/local/Cellar/go/1.7.4/libexec/src/reflect/value.go:302 +0xa4
2016/12/21 00:21:24 packer: net/rpc.(*service).call(0xc42025e700, 0xc42025e6c0, 0xc420160618, 0xc420262780, 0xc42025afa0, 0xf17440, 0xc4201d19dc, 0x18a, 0xeec040, 0xc4201d19e0, ...)
2016/12/21 00:21:24 packer: 	/usr/local/Cellar/go/1.7.4/libexec/src/net/rpc/server.go:383 +0x148
2016/12/21 00:21:24 packer: created by net/rpc.(*Server).ServeCodec
2016/12/21 00:21:24 packer: 	/usr/local/Cellar/go/1.7.4/libexec/src/net/rpc/server.go:477 +0x421
2016/12/21 00:21:24 ui error: Build 'amazon-ebs' errored: unexpected EOF
2016/12/21 00:21:24 Builds completed. Waiting on interrupt barrier...
2016/12/21 00:21:24 machine readable: error-count []string{"1"}
2016/12/21 00:21:24 ui error:
==> Some builds didn't complete successfully and had errors:
2016/12/21 00:21:24 machine readable: amazon-ebs,error []string{"unexpected EOF"}
2016/12/21 00:21:24 ui error: --> amazon-ebs: unexpected EOF
2016/12/21 00:21:24 ui:
==> Builds finished but no artifacts were created.
2016/12/21 00:21:24 waiting for all plugin processes to complete...
2016/12/21 00:21:24 /usr/local/Cellar/packer/0.12.1/bin/packer: plugin process exited
2016/12/21 00:21:24 /usr/local/Cellar/packer/0.12.1/bin/packer: plugin process exited
2016/12/21 00:21:24 /usr/local/Cellar/packer/0.12.1/bin/packer: plugin process exited
2016/12/21 00:21:24 /usr/local/Cellar/packer/0.12.1/bin/packer: plugin process exited
```

We're using IAM roles, but I tried it with a full privs role with the same result.  On the other hand, it's only crashing on a EBS builder with 2 x volumes.  I've successfully built a 1 x EBS volume AMI twice, and it has twice failed trying to build 2 x volumes.

--- a/builder/amazon/ebs/step_create_ami.go
+++ b/builder/amazon/ebs/step_create_ami.go
@@ -68,7 +68,8 @@ func (s *stepCreateAMI) Run(state multistep.StateBag) multistep.StepAction {
 
 	snapshots := make(map[string][]string)
 	for _, blockDeviceMapping := range imagesResp.Images[0].BlockDeviceMappings {
-		if blockDeviceMapping.Ebs != nil {
+		if blockDeviceMapping.Ebs != nil && blockDeviceMapping.Ebs.SnapshotId != nil {
+
 			snapshots[*ec2conn.Config.Region] = append(snapshots[*ec2conn.Config.Region], *blockDeviceMapping.Ebs.SnapshotId)
 		}
 	}

